### PR TITLE
Use PyPI RSS feed to fill in missing published dates

### PR DESCRIPTION
**Pull request by johnbintz-tidelift**
This breaks out much of the data retrieval code for the PyPI package manager, centralizes version processing in a set of objects, and adds the ability to load published dates from a package's RSS feed.

### CODE REVIEW

**Review by macowie — APPROVED** [review]

**Review by djpowers — APPROVED** [review]

File: app/models/package_manager/pypi/json_api_project.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          homepage_url
+        )
+      end
+
+      def releases
+        JsonApiProjectReleases.new(
+          @data["releases"].map do |version_number, details|
+            JsonApiProjectRelease.new(
+              version_number: version_number,
+              published_at: self.class.release_data_published_at(details)
```
Comment (jhan217) [inline app/models/package_manager/pypi/json_api_project.rb:89]: Could `release_data_published_at` be an instance method? I only see it being used here

File: app/models/package_manager/pypi/version_processor.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        @project_releases.map do |project_release|
+          next @known_versions[project_release.version_number] if @known_versions.key?(project_release.version_number)
+
+          version_number = project_release.version_number
+          original_license = json_api_single_release_for_version(version_number).license
+          rss_api_release = rss_api_releases_hash[version_number]
+          published_at = project_release.published_at || rss_api_release&.published_at
+
+          {
+            number: project_release.version_number,
```
Comment (jhan217) [inline app/models/package_manager/pypi/version_processor.rb:20]: Could this just be `version_number`?

**Review by jhan217 — APPROVED** [review]
LGTM -- thanks for modularizing all of this, hopefully it will be easier to build on in the future! :)

File: app/models/package_manager/pypi/json_api_project.rb (hunk 1 above)
Comment (johnbintz-tidelift) [reply app/models/package_manager/pypi/json_api_project.rb:89]: I would want to make it a private method and then add tests to `#releases` to ensure all the private method's cases are handled.

Comment (johnbintz-tidelift) [reply app/models/package_manager/pypi/json_api_project.rb:89]: Scratch that, this is going to need more refactoring to be structured correctly.

**Review by jhan217 — APPROVED** [review]

**Pull request merged**